src/darwin/frida-helper-backend-glue.m

```diff
@@ -1450,6 +1466,74 @@ static void frida_darwin_helper_backend_launch_using_sbs (NSString * identifier,
   return killed_pid;
 }
 
+static gboolean
+frida_has_active_prewarm (gint pid)
+{
+  gboolean result = false;
+  int mib_argmax[] = { CTL_KERN, KERN_ARGMAX };
+  int mib_args[] = { CTL_KERN, KERN_PROCARGS2, 0 };
+  size_t size;
+  gint32 arg_max = 0, argc;
+  guint8 * buffer = NULL, * cursor, * end;
```

Comment:
Wonder if it would be more readable to move the assignment part down to just before where `size` is assigned. Since `buffer` is what this logical unit of the function is going to create, we could move it before `size` in the variable declarations, and assign it first, to kind of say "here we're about to compute *buffer*".